Sources/Subs-Attachments.php

```diff
@@ -1346,4 +1348,50 @@ function loadAttachmentContext($id_msg, $attachments)
 	return $attachmentData;
 }
 
+/**
+ * prepare the Attachment api for all messages
+ *
+ * @param int array $msgIDs the message ID to load info from.
+ *
+ * @return void.
+ */
+function prepareAttachsByMsg($msgIDs)
+{
+	global $context, $modSettings, $smcFunc, $user_info, $cacheMsgID;
+	if(!isset($cacheMsgID))
+		$cacheMsgID = array();
+
+	// remove all $msgIDs which we already cached
+	$msgIDs = array_flip(array_diff_key(array_flip($msgIDs), $cacheMsgID)) ;
```

Comment:
This works, but it would be more efficient to use the following.
```
$msgIDs = array_diff($msgIDs, array_keys($cacheMsgID));
```

Also, you could put an `else` before this line, because there is no need to do it at all if `$cacheMsgID` is empty.

Reply:
``$msgIDs = array_diff($msgIDs, array_keys($cacheMsgID));``
since the $cacheMsgID is possibe bigger of memory footprint and got a more complex structur,
i choose the other way.